### PR TITLE
Set CMP0115 to new

### PR DESCRIPTION
**Pull request by Pospelove**
Require explicit extensions for source files, do not guess.
The extra calls to GetFileAttributesW significantly slow down cmake on Windows. https://gitlab.kitware.com/cmake/cmake/-/issues/23154

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,6 +3,10 @@ cmake_minimum_required(VERSION 3.15) # 3.15 for MSVC_RUNTIME_LIBRARY, removing d
 cmake_policy(SET CMP0091 NEW)
 cmake_policy(SET CMP0092 NEW)
 
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.20)
```
Comment (Neumann-A) [inline CMakeLists.txt:6]: Is there a reason not to use `if(POLICY CMP0115)`

Comment (Neumann-A) [reply CMakeLists.txt:6]: Also i think vcpkg could simply do:
`cmake_minimum_required(VERSION 3.15...3.24)`

Comment (Neumann-A) [reply CMakeLists.txt]: I think you can then remove all the explicit settings of `cmake_policy`

**Review by BillyONeal — APPROVED** [review]
LGTM with Neumann-A's suggested change of removing explicit policy settings applied.

**Comment by dg0yt** [comment]
> Should we also be taking steps to try to upgrade our helper ports where we don't much care about CMake version for this?

Not soure what is "this" but CMP0115 doesn't seem relevant for CMake script mode.

**Comment by BillyONeal** [comment]
> > Should we also be taking steps to try to upgrade our helper ports where we don't much care about CMake version for this?
> 
> Not soure what is "this" but CMP0115 doesn't seem relevant for CMake script mode.

I mean things where we've provided a replacement build system. For example, https://github.com/microsoft/vcpkg/blob/e2667a41fc2fc578474e9521d7eb90b769569c83/ports/lua/CMakeLists.txt#L13

**Review by BillyONeal — APPROVED** [review]
Will merge when build comes back. Thanks for your contribution!

**Comment by BillyONeal** [comment]
Build break is from stealth merge conflict, fixed in https://github.com/microsoft/vcpkg-tool/pull/744

**Comment by BillyONeal** [comment]
@Neumann-A It looks like the explicit settings are required. @Pospelove Sorry for the noise!

**Comment by Neumann-A** [comment]
@BillyONeal I disagree:

Using:
```
cmake_minimum_required(VERSION 3.12...3.25) # This is intentionally higher than 3.24 to see what the behavior is if it is higher than the installed version
project(policy_test LANGUAGES NONE)
cmake_policy(GET CMP0115 VAR)
message(STATUS "VAR:${VAR}")
```
I get: 
```
E:\vcpkg_folders\cmake_policy>cmake -G "Visual Studio 17 2022" -B build -S .
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19044.
-- VAR:NEW
-- Configuring done
-- Generating done
-- Build files have been written to: E:/vcpkg_folders/cmake_policy/build
```

**Comment by BillyONeal** [comment]
> @BillyONeal I disagree:
> 
> 
> 
> Using:
> 
> ```
> 
> cmake_minimum_required(VERSION 3.12...3.25) # This is intentionally higher than 3.24 to see what the behavior is if it is higher than the installed version
> 
> project(policy_test LANGUAGES NONE)
> 
> cmake_policy(GET CMP0115 VAR) # Added in 3.20 so if it uses 3.12 policy it should be OLD or unset.
> 
> message(STATUS "VAR:${VAR}")
> 
> ```
> 
> I get: 
> 
> ```
> 
> E:\vcpkg_folders\cmake_policy>cmake -G "Visual Studio 17 2022" -B build -S .
> 
> -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19044.
> 
> -- VAR:NEW
> 
> -- Configuring done
> 
> -- Generating done
> 
> -- Build files have been written to: E:/vcpkg_folders/cmake_policy/build
> 
> ```

I based this statement off the build failing: https://dev.azure.com/vcpkg/public/_build/results?buildId=79160&view=results

Is the high number exclusive?

**Comment by Neumann-A** [comment]
You probably need 3.24.2 instead of 3.24(.0) for that

**Pull request merged**